API.md

```diff
@@ -0,0 +1,121 @@
+API Proposal
+============
+## Use Cases
+- **Application Administration** (e.g. GPII): updating multiple settings of multiple (other) applications
+- **Application Settings** (e.g. my app): read/update/monitor applications's own settings
+
+
+## To throw or not to throw
+
+## Module Styles
+[Interesting read-up concerning nodejs module design](http://psteeleidem.com/node-module-factory-pattern/)
+
+### 1) Module style - Namespaced Collection of functions
```

Comment:
+1

Also IMO, and in terms of naming convention for functions, it would be great to replicate the names in the command-line utility

```
  list-schemas              List installed schemas
  list-relocatable-schemas  List relocatable schemas
  list-keys                 List keys in a schema
  list-children             List children of a schema
  list-recursively          List keys and values, recursively
  range                     Queries the range of a key
  get                       Get the value of a key
  set                       Set the value of a key
  reset                     Reset the value of a key
  reset-recursively         Reset all values in a given schema
  writable                  Check if a key is writable
  monitor                   Watch for changes
```